rama-http/src/layer/auth/require_authorization.rs

```diff
@@ -258,6 +260,53 @@ where
     }
 }
 
+pub(crate) struct AuthorizeContext<C> {
```

Comment:
This winds up leaking into the public interface

Reply:
```suggestion
pub struct AuthorizeContext<C> {
```

You're correct. Just make this `pub(crate) struct AuthorizeContext<C>`. It's still private in creation as it cannot be created outside of the crate, but at least it lets you name the type :)